org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaDerivedStateComputer.java

```diff
@@ -126,6 +126,9 @@ public JvmDeclaredType createType(TypeDeclaration type, String packageName) {
 		case TypeDeclaration.ANNOTATION_TYPE_DECL:
 			jvmType = TypesFactory.eINSTANCE.createJvmAnnotationType();
 			break;
+		case 5 /*TypeDeclaration.RECORD_DECL*/:
```

Comment:
see also discussion in

https://github.com/eclipse/xtext/issues/2823
and 
https://github.com/eclipse/xtext/pull/3037

Reply:
should we use the preview constant? it seems to be in 33 too

Reply:
I would use `TypeDeclaration.RECORD_DECL`: we already use other internal constants; it should not be a preview feature anymore in Java 21; maybe they forgot to remove the comment? Or maybe they refer to previous Java versions.

I don't understand the other comment. You're not talking about specifically about records, but in general about the `default` case? Maybe it's better to default to generic type as we do in JdtBasedTypeFactory, though I don't understand the part about super type.

Reply:
the jdtbasedfactory does not throw. it falls back to jvm generic type

Reply:
i am asking if we should do it here to. any idea @szarnekow